### PR TITLE
CADC-8616 Updated renaming algorithm and corresponding tests.

### PR DESCRIPTION
**Pull request by yeunga**
In addition to caom2-tap, the following libraries and services are affected by this change and will need to be recompiled, published and released.
Libraries:
caom2-soda-server
caom2-pkg-server
caom2-meta-server
caom2-tap-server
caom2-datalink-server
sc2links
sc2meta
sc2soda
sc2tap

Services:
caom2ops
ams
argus
tap

### CODE REVIEW

File: caom2-tap/src/main/java/ca/nrc/cadc/caom2ops/AdCutoutGenerator.java [hunk 1]
```diff
@@ -89,16 +89,38 @@ static String removeCompressionExtension(String uriFilename) {
...
+                    String uncompressedFilename = removeCompressionExtension(ssp.substring(i + 1));
+                    String head = uncompressedFilename;
+                    String tail = "";
+                    int index = uncompressedFilename.lastIndexOf('.');
+                    if (index > -1) {
+                        head = uncompressedFilename.substring(0, index);
+                        tail = uncompressedFilename.substring(index, uncompressedFilename.length());
+                    }
+
+                    filename = head + "__" + label + "__" + replaceNonAlphanumeric(cutouts) + tail;
```
Comment (hjeeves) [inline caom2-tap/src/main/java/ca/nrc/cadc/caom2ops/AdCutoutGenerator.java:123]: replaceNonAlphanumeric(cutouts) can be "", and tail as well, so can filename have '__ ' as it's ending? Does that matter?

Comment (hjeeves) [reply caom2-tap/src/main/java/ca/nrc/cadc/caom2ops/AdCutoutGenerator.java:123]: Is there a limit to the filename length that could be exceeded if 'head' is somehow too long?

**Review by hjeeves — APPROVED** [review]

**Comment by pdowler** [comment]
Why do all those libraries and services need to be rebuilt? I would expect that caom2ops and sc2soda need to be released as they are the only services that invoke a CutoutGenerator (with a cutout string). 

caom2-tap-server doesn't even depend on caom2-tap (in fact, you might be surprised to note that it doesn't depend on *any* caom2 code)

**Comment by yeunga** [comment]
I just thought that we want to have one version caom2-tap being used in production. It's true that we can limit our updates to only caom2ops and sc2soda.

You are right. caom2-tap-server does not need to be recompiled because it does not depend on caom2-tap.

**Review by pdowler — APPROVED** [review]

**Pull request merged**